src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int capacity = DEFAULT_CAPACITY;
```

Comment:
do we need this field?

Reply:
don`t know exact reason, but yes:
If I don`t initialize capacity here instead of  constructor, many tests ruins, and i don`t understand why... Also i using DEFAULT_CAPACITY, not number directly because of checklist: 
Default capacity of the array should be imaged as a constant field.

Reply:
Found how to deal with